test/functional/p2p_ibd_txrelay.py

```diff
@@ -22,7 +41,36 @@ def set_test_params(self):
             ["-minrelaytxfee={}".format(NORMAL_FEE_FILTER)],
         ]
 
-    def run_test(self):
+    def test_inv_ibd(self):
+        self.log.info("Check that nodes don't request txn while still in IBD")
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
+        peer.send_message(msg_inv([CInv(t=MSG_WTX, h=0xffaa)]))
+        # As tx-announcing connection is inbound, wait for the inbound delay applied by requester.
+        sleep(NONPREF_PEER_TX_DELAY)
+        peer.sync_with_ping()
+        with p2p_lock:
+            assert_equal(peer.tx_getdata_count, 0)
+        peer.peer_disconnect()
+        peer.wait_for_disconnect()
+
+    def test_unrequested_tx_ibd(self):
+        self.log.info("Check that nodes don't process unrequested txn while still in IBD")
+
+        unrequested_tx = CTransaction()
+        unrequested_tx.vin.append(CTxIn())
+
+        unrequested_tx.vout.append(CTxOut())
+        unrequested_tx.calc_sha256()
+
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
+        peer.send_message(msg_tx(unrequested_tx))
+        with self.nodes[0].assert_debug_log(expected_msgs=['unrequested transaction while in IBD from peer=2'], unexpected_msgs=['from peer=2 was not accepted: scriptpubkey']):
```

Comment:
The unexpected_msgs here is very specific. I understand that you're trying to check that the tx wasn't rejected by the mempool, but if the log format ever changes, then the unexpected_msgs string isn't actually testing anything, and would just be dead code. I'd suggest removing it and relying on the expected_msgs string.

Reply:
Thanks for the tip, removed it. I've searched for a while for this but I don't think we have better than checking the logs to test that feeding the node with some input (an unrequested tx) is effectively rejected by the targeted code check (the new check L2928) and not by the mempool checks further down in the code path ?

I'm still ramping up on our p2p functional test framework :)